src/sagemaker/cli/framework_upgrade.py

```diff
@@ -0,0 +1,284 @@
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2)
+        image_type (str): Framework image type, it could be "training", "inference"
+            or "eia"
+        processors (list): Supported processors (e.g. ["cpu", "gpu"])
+        py_versions (list): Supported Python versions (e.g. ["py3", "py37"])
+        registries (dict): Framework image's region to account mapping.
+        repository (str): Framework image's ECR repository.
+    """
+    for processor in processors:
+        if processor not in existing_content[image_type]["processors"]:
+            existing_content[image_type]["processors"].append(processor)
+    existing_content[image_type]["version_aliases"][short_version] = full_version
+
+    add_version = {
+        "registries": registries,
+        "repository": repository,
+        "py_versions": py_versions,
+    }
+    existing_content[image_type]["versions"][full_version] = add_version
+
+
+def add_algo_version(
+    existing_content,
+    processors,
+    scopes,
+    full_version,
+    py_versions,
+    registries,
+    repository,
+    tag_prefix,
+):  # pylint: disable=W0621
+    """Update Algorithm image uri json file with new version information.
+
+    Args:
+        existing_content (dict): Existing algorithm image uri information read from
+            "<algorithm>.json" file.
+        processors (list): Supported processors (e.g. ["cpu", "gpu"])
+        scopes (list): Framework image type, it could be "training", "inference
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2)
+        py_versions (list): Supported Python versions (e.g. ["py3", "py37"])
+        registries (dict): Algorithm image's region to account mapping.
+        repository (str): Algorithm's corresponding repository name.
+        tag_prefix (str): Algorithm image's tag prefix.
+    """
+    for processor in processors:
+        if processor not in existing_content["processors"]:
+            existing_content["processors"].append(processor)
+    for scope in scopes:
+        if scope not in existing_content["scope"]:
+            existing_content["scope"].append(scope)
+
+    add_version = {
+        "py_versions": py_versions,
+        "registries": registries,
+        "repository": repository,
+    }
+    if tag_prefix:
+        add_version["tag_prefix"] = tag_prefix
+    existing_content["versions"][full_version] = add_version
+
+
+def add_region(framework, existing_content, region, account):  # pylint: disable=W0621
+    """Add region account to framework/algorithm registries.
+
+    Args:
+        framework (str): Framework name (e.g. tensorflow, pytorch, mxnet).
+        existing_content (dict): Existing framework/algorithm image uri information read from
+            json file.
+        region (str): New region to be added to framework/algorithm registries (e.g. us-west-2).
+        account (str): Region registry account number.
+    """
+    if framework in DLC_FRAMEWORK:
```

Comment:
rather than hardcoding frameworks names, I think I'd rather there be a check `if scope in existing_content` sort of thing

Reply:
updated the code to use "scope" as key to differentiate DLC framework and algorithms